### PR TITLE
ENH: add axisbelow option 'line', make it the default

### PR DESCRIPTION
**Pull request by efiring**
The present `axisbelow` value can be only True, which puts the ticks and gridlines below patches, or False, which puts them above lines.  It might make more sense to put them above patches (e.g. filled contours) and below lines.  This PR adds that option, and makes it the default.  See #5980 and #4243.

Ideally we would be able to control ticks and gridlines separately, as requested in #4243, but this would require major refactoring.  Ticklines and gridlines are drawn by `Tick.draw` which is called by `Axis.draw`.  Hence the order in which they are drawn relative to other plot elements is controlled by `Axes.xaxis.zorder` etc., not by the zorder of each gridline, for example.

### CODE REVIEW

File: lib/matplotlib/rcsetup.py [hunk 1]
```diff
@@ -981,7 +992,7 @@ def validate_animation_writer_path(p):
     'errorbar.capsize':      [0, validate_float],
 
     # axes props
-    'axes.axisbelow':        [False, validate_bool],
+    'axes.axisbelow':        ['line', validate_axisbelow],
```
Comment (tacaswell) [inline lib/matplotlib/rcsetup.py]: also need to change in the template

**Comment by tacaswell** [comment]
I am moderately :+1: on this.  Would it be better to let the rcparam be a float ?

**Comment by efiring** [comment]
I thought about the float option, but using strings fits better with the (perhaps unfortunate) existing name of the parameter/kwarg.  "axisbelow line" makes a little bit of sense; "axisbelow 1.5" doesn't.
I originally took a stab at setting zorder of the gridlines, but it didn't work, for the reason given in the commit comment above.

**Comment by efiring** [comment]
There are 3 image failures involving minuscule differences.  In the one I reproduced, I can't even see anything other than solid black in the diff image.  I would not expect this PR to affect any images because the classic style is unchanged.

**Comment by WeatherGod** [comment]
I have been seeing tiny image comparison failures locally. The diff images
show me all black, too.

**Comment by tacaswell** [comment]
did some of the pgf tests avoid getting run in 'classic' mode?

**Comment by jenshnielsen** [comment]
Yes the pgf tests have their own slightly different image comparison for some reason. They will not run in classic mode

**Comment by efiring** [comment]
The one Travis failure is in the docs build.  I don't understand it.

**Comment by efiring** [comment]
The Travis failure is in the docs build, ./examples/api/custom_scale_example.py.  That's a pretty complicated example, but I don't see how its failure could be related to the present PR. @mdboom, any ideas?

**Comment by efiring** [comment]
The example fails with `rcParams['axes.axisbelow'] = True` as well as with the new default, so it has always been broken--this PR is just triggering the failure.

**Comment by efiring** [comment]
I think this is complete now.  The last thing I added is a note in whats_new/style_changes.  The change is visually so subtle that maybe it doesn't need to be mentioned here; if that's the case, the last commit can be removed.

**Comment by mdboom** [comment]
:+1: after adding a test.

**Comment by tacaswell** [comment]
backported to v2.x as 2ec3780

**Pull request merged**